### DISCUSSION TITLE
Make Python read_parquet() (and read_json()) support file object as input like read_csv() does

### DISCUSSION THREAD

**Discussion by daviddwlee84** [duckdb/duckdb#9857] [Ideas]
Is it possible to make `read_parquet()` able to accept file objects (e.g. `open()`, `io.BytesIO`, ...) as input.
It is useful when making an online application that uploads a file.
For example, using [Streamlit's `file_uploader`](https://docs.streamlit.io/library/api-reference/widgets/st.file_uploader).

Here is a simple demonstration:

```python
import duckdb
import shutil

# Getting uploaded_file by opening a file or uploading a file
uploaded_file = ...

# CSV example
# uploaded_file is an opened CSV file object
data = duckdb.read_csv(uploaded_file)

# Parquet workaround
# uploaded_file is an opened Parquet file object
with open(uploaded_file.name, 'wb') as destination_file:
    shutil.copyfileobj(uploaded_file, destination_file)
data = duckdb.read_parquet(uploaded_file.name)

# What I expected for parquet experience
data = duckdb.read_parquet(uploaded_file)
```

---

The `read_parquet()` function signature

https://github.com/duckdb/duckdb/blob/9c91b3a329073ea1767b0aaff94b51da98dd03e2/tools/pythonpkg/duckdb-stubs/__init__.pyi#L269-L272

The `read_csv()` function signature

https://github.com/duckdb/duckdb/blob/9c91b3a329073ea1767b0aaff94b51da98dd03e2/tools/pythonpkg/duckdb-stubs/__init__.pyi#L222-L224

The `read_json()` function signature

https://github.com/duckdb/duckdb/blob/9c91b3a329073ea1767b0aaff94b51da98dd03e2/tools/pythonpkg/duckdb-stubs/__init__.pyi#L589-L590

**Comment by daviddwlee84** [comment]
Another workaround

```python
import pandas as pd

# Parquet workaround
# uploaded_file is an opened Parquet file object
data = duckdb.from_df(pd.read_parquet(uploaded_file))
```

> Found another temp file workaround.
> 
> [Get path from file_uploader() - 🎈 Using Streamlit - Streamlit](https://discuss.streamlit.io/t/get-path-from-file-uploader/3771/17)
> 
> But seems `duckdb.read_parquet` is lazy loading, you will have to keep the temp file until you no longer need it.

**Comment by Djfe** [comment]
Ever since v1.1.0 **DuckDB for Python** has the path or buffer syntax now for `read_json`
https://github.com/duckdb/duckdb/commit/34207253b4175f786c525df1527945ae1196b104

I guess, the developers want/prefer the file_glob syntax for parquet or Noone bothered to write the change, yet
